### PR TITLE
[Prefixed Rector] Add package-scoper approach

### LINKED ISSUE #4406: Could not process file due to BetterReflection\ReflectionClass "Carbon\Traits\Localization" could not be found in the located source (with POC)
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | Rector 0.8.23              |
| Installed as   | rector/rector-prefixed                      |

Somehow Rector trips when using [Carbon](https://github.com/briannesbitt/Carbon) together with `AUTO_IMPORT_NAMES`.

```
$ vendor/bin/rector process src --dry-run
Rector 0.8.23
Config file: rector.php

 4/4 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

1 file with changes
===================

1) src/Entity.php

    ---------- begin diff ----------
--- Original
+++ New
@@ -4,13 +4,11 @@

 namespace MyNamespace;

-use Carbon\Carbon;
-
+use Carbon;
 class Entity
 {
     private Carbon $createdAt;
-
-    public function __construct()
+    public function __construct()
     {
         $this->createdAt = Carbon::now();
     }
    ----------- end diff -----------



 [ERROR] Could not process "src/Entity.php" file, due to:
         "_HumbugBox2e5f015a1738\Roave\BetterReflection\Reflection\ReflectionClass "Carbon\Traits\Localization" could
         not be found in the located source".

```

## Minimal PHP Code Causing Issue

Please see working POC repository here: https://github.com/ruudk/rector-carbon-bug

## Expected Behaviour

Not crash.

### PR DESCRIPTION
**Pull request by TomasVotruba**
Pre-requisite for https://github.com/rectorphp/rector/issues/4913

Closes https://github.com/rectorphp/rector/issues/4406

<br>

## Reasoning

Why the switch from phar to scoped package? See https://github.com/symplify/symplify/issues/2465#issuecomment-723644671

tl;dr; This allows release of downgraded PHP 7.1, 7.0, 5.6 version. So Rector will not need Docker to run on the most legacy systems.	

## Todo

- push code here: https://github.com/rectorphp/rector-prefixed
- the CI there must pass

### CODE REVIEW

File: .github/workflows/build_scoped_rector.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                uses: symplify/github-action-monorepo-split@master
+                env:
+                    GITHUB_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+                with:
+                    package-directory: 'rector-scoped'
+                    split-repository-organization: 'rectorphp'
+                    split-repository-name: 'rector-prefixed'
+                    tag: ${{ steps.previous_tag.outputs.tag }}
+                    user-name: "kaizen-ci"
+                    user-email: "info@kaizen-ci.org"
```
Comment (staabm) [inline .github/workflows/build_scoped_rector.yml:90]: ohh I see why you asked this question about kaizen on twitter ;-)

Comment (TomasVotruba) [reply .github/workflows/build_scoped_rector.yml:90]: Also ;)... I was exploring if more developers know Japanese :)

**Comment by TomasVotruba** [comment]
Hi @escopecz, good news! You might like this one, new scoped package is ready to be tested :)
Could you give a try? I bet there are some spots we missed

File: build-rector-scoped.sh [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+note "Coping root files to $NESTED_DIRECTORY directory"
+rsync -av * "$NESTED_DIRECTORY" --quiet
+
+note "Running composer update without dev"
+composer update --no-dev --no-progress --ansi --working-dir "$NESTED_DIRECTORY" # --ignore-platform-req php
+
+# Unpacking PHPStan
+note "Unpacking PHPStan"
+wget https://github.com/box-project/box/releases/download/3.11.0/box.phar -N --no-verbose
+php box.phar extract "$NESTED_DIRECTORY/vendor/phpstan/phpstan/phpstan.phar" "$NESTED_DIRECTORY/vendor/phpstan/phpstan-extracted"
```
Comment (TomasVotruba) [inline build-rector-scoped.sh:43]: @ondrejmirtes Hi Ondra, I just want to thank you for the idea of unpacking `phpstan.phar` and using it's code instead of using `phpstan/phpstan-src`. It saved me so much troubles with duplicating your process. 

Now it rector prefixing works well and `scoper.php.inc` is thin as snow layer :)

I never though of that being a possibility until you told me. :bow:

**Comment by ondrejmirtes** [comment]
Nice 😊 You're welcome.
